[WIP] Update Install Guide for v4.0.0 (#169)
libremfg/rhize-docs: content/deploy/install/services.md

For the full configuration options,
read the official [Helm `install` reference](https://helm.sh/docs/helm/helm_install/).

### Redpanda

Rhize uses Redpanda to buffer requests to Restate and connect to Agent.

Install Redpanda with these steps:

> 1. If it doesn't exist, add the Redpanda repository:

Reviewer: ```suggestion
1. If the Redpanda repository doesn't exist, add it:
```

    ```

1. Modify the Helm file as needed.

1. Install with Helm:

    ```bash
    helm install tempo -f tempo.yaml grafana/tempo -n {{< param application_name >}}
    ```

> > Note: Depending on your configuration you may need to run Tempo in distributed mode. When installing with Helm instead of using `grafana/tempo` instead do `grafana/tempo-distributed`.

Reviewer: ```suggestion
> [!NOTE]
> Depending on your configuration you may need to run Tempo in distributed mode. When installing with Helm instead of using `grafana/tempo` instead do `grafana/tempo-distributed`.
```

### Restate

Install Restate with these steps:

1. Modify the Helm file as needed.

1. Install with Helm:

    ```bash
    helm install restate -f restate.yaml oci://ghcr.io/restatedev/restate-helm -n {{< param application_name >}}
    ```

> The port for Restate will need to be proxied in order to register certain services with it.

Reviewer: ```suggestion
So that you can register certain services with restate, proxy the Restate port:
```

   ```bash
   kubectl port-forward -n {{< param application_name >}} pod/restate-0 9070:9070
   ```

### Workflow

The Workflow service is the custom engine Rhize uses to process low-code workflows modeled in the Workflow UI.

> **Requirements**: The Workflow service requires the [{{< param db >}}](#db), [Restate](#restate), and [Tempo](#tempo) services.

Install Workflow with these steps:

1. Modify the Helm file as needed.

1. Install with Helm:

   ```bash
   helm install workflow -f workflow.yaml {{< param application_name >}}/workflow -n {{< param application_name >}}
   ```

> 1. Workflow should register with Restate when it starts up. If it doesn't then it can be registered it by running the following:

Reviewer: ```suggestion
1. When the Workflow service starts, it should register with Restate. If you need to manually register, run the following command:
```

    ```bash
    curl --location 'http://localhost:9070/deployments' \
      --header 'Content-Type: application/json' \
      --data '{"uri":"http://workflow.{{< param application_name >}}.svc.cluster.local:29080", "force":true}'
    ```

### Typescript Host Service

Install Typescript Host Service with these steps:

1. Modify the Helm file as needed.

1. Install with Helm:

   ```bash
   helm install typescript-host-service -f typescript-host-service.yaml {{< param application_name >}}/typescript-host-service -n {{< param application_name >}}
   ```

> 1. Register with Restate

Reviewer: ```suggestion
1. Register with Restate:
```

    ```bash
    curl --location 'http://localhost:9070/deployments' \
      --header 'Content-Type: application/json' \
      --data '{"uri":"http://typescript-host-service.{{< param application_name >}}.svc.cluster.local:9081", "force":true}'
    ```

### QuestDB

Install QuestDB with these steps:

1. If it doesn't exist, add the Redpanda repository:

    ```bash
    helm repo add questdb https://helm.questdb.io/
    helm repo update
    ```

1. Modify the Helm file as needed.

1. Install with Helm:

    ```bash
    helm install questdb -f questdb.yaml questdb/questdb -n {{< param application_name >}}
    ```

> ### ISA 95
> 
> Install ISA 95 with these steps:
> 
> 1. Modify the Helm file as needed.
> 
> 1. Install with Helm:

Reviewer: ```suggestion
### ISA-95

Install the ISA-95 service with these steps:

1. Modify the Helm file as needed.

1. Install with Helm:
```

   ```bash
   helm install isa95 -f isa95.yaml {{< param application_name >}}/isa95 -n {{< param application_name >}}
   ```

> 1. ISA 95 should register with Restate when it starts up. If it doesn't then it can be registered it by running the following:

Reviewer: ```suggestion
1. ISA-95 should register with Restate when it starts up.
If it doesn't, run the following:
```

The Admin UI is the graphical frontend to [handle events]({{< relref "/how-to/bpmn" >}}) and [define work masters]({{< relref "/how-to/model" >}}).

> **Requirements:** The Admin UI requires the [Workflow](#workflow) services.

After installing all other services, install the UI with these steps:

1. Modify the UI Helm file as needed.

1. Install with Helm:

    ```bash
    helm install admin-ui -f admin-ui.yaml {{< param application_name >}}/admin-ui -n {{< param application_name >}}
    ```

If the install is successful, the UI is available on its
[default port]({{< ref "default-ports" >}}).

## Optional: Agent

Install Agent Service with these steps:

1. Modify the Agent Helm file as needed.

1. In Rhize add in a Data Source for Agent to interact with:
    - In the lefthand menu open `Master Data` > `Data Sources` > `+ Create Data Source`
    - Input a name for the Data Source.
    - Add a Connection String and Create.
    - Add any relevant Topics.
    - Activate the Data Source.

1. Install with Helm:

    ```bash
    helm install agent -f agent.yaml {{< param application_name >}}/agent -n {{< param application_name >}}
    ```

> Agent can be verified to be working by checking in Redpanda's UI.

Reviewer: ```suggestion
To verify that Agent is working, check the Redpanda UI.
```
